### ISSUE TITLE
Add-on module crashes when emitting close events during garbage collection of a wrapped object

### ISSUE THREAD

**Issue by stephenwvickers** [stephenwvickers/node-raw-socket#3]

**Comment by stephenwvickers** [comment]
```diff
# HG changeset patch
# User Kazuki Yamaguchi <k@rhe.jp>
# Date 1440307350 -32400
# Node ID fd19093ac63ae8c370bd3d9d8c13c3dd25329598
# Parent  8d470c76f8640354a11a7324876bf0aca0db3829
Don't emit "close" when closed by garbage collector since callback function no longer exists

diff --git a/src/raw.cc b/src/raw.cc
--- a/src/raw.cc
+++ b/src/raw.cc
@@ -300,13 +300,15 @@
 		this->poll_initialised_ = false;
 	}
 
-	Local<Value> emit = NanObjectWrapHandle(this)->Get(NanNew<String>("emit"));
-	Local<Function> cb = emit.As<Function> ();
+	if (!this->deconstructing_) {
+		Local<Value> emit = NanObjectWrapHandle(this)->Get(NanNew<String>("emit"));
+		Local<Function> cb = emit.As<Function> ();
 
-	Local<Value> args[1];
-	args[0] = NanNew<String>("close");
+		Local<Value> args[1];
+		args[0] = NanNew<String>("close");
 
-	cb->Call (NanObjectWrapHandle(this), 1, args);
+		cb->Call (NanObjectWrapHandle(this), 1, args);
+	}
 }
 
 int SocketWrap::CreateSocket (void) {
```

**Comment by stephenwvickers** [comment]
Applied this patch, will be available when I push 1.4.0 to npm soon.
